### PR TITLE
Add documentation about job-states

### PR DESCRIPTION
**Pull request by romandecker**
Adds a chart and some textual explanation to the usage docs about the various job states and their transitions as discussed in #206. Also includes a raw [`.drawio`](http://draw.io/) file to have an editable copy of the diagram around.

Feel free to edit the docs text however you like, I kinda went freestyle there ¯\\\_(ツ)\_/¯

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- TOC -->
 
 - [Intro](#intro)
+  - [`error`](#error)
```
Comment (timgit) [inline docs/usage.md:6]: toc issue

File: docs/usage.md [hunk 2]
```diff
@@ -59,6 +60,16 @@ pg-boss is used by creating an instance of the exported class, a subclass of a N
...
+## Job states
+
+A pg-boss job undergoes certain states during its lifetime. All jobs start out as `created` - from there, they will usually become `active` when picked up for work. If job processing completes successfully, jobs will go to `completed`. If job processing is not successful, jobs will go to either `failed` or `retry` (if they were started with the respective retry options). It's also possible for `active` jobs to become `expired`, which happens when job processing takes too long.
+
+Jobs can also be `cancelled` via [`cancel(id)`](#cancelid) or [`cancel([ids])`](#cancelids), which will transition them into the `cancelled` state.
+
+All jobs that are `completed`, `expired`, `cancelled` or `failed` become eligible for archiving (i.e. they will transition into the `archive` state) after the configured `archiveCompletedAfterSeconds` time. Once `archive`d, jobs will be automatically deleted by pg-boss after the configured deletion period.
+
+Here's a state diagram that shows the possible states and their transitions:
+
```
Comment (timgit) [inline docs/usage.md:72]: todo: insert markdown to embed image

**Review by timgit — CHANGES_REQUESTED** [review]

File: docs/usage.md (hunk 2 above)
Comment (romandecker) [reply docs/usage.md:72]: Hm, must've screwed up the last commit - sorry about that ¯\_(ツ)_/¯

df69abc Fix TOC & job states image in usage docs

File: docs/usage.md (hunk 1 above)
Comment (romandecker) [reply docs/usage.md]: df69abc Fix TOC & job states image in usage docs

**Comment by romandecker** [comment]
Sorry about that - should be fixed now.

**Comment by timgit** [comment]
thanks!

**Pull request merged**